Sources/Overlays/_Testing_WinSDK/Attachments/AttachableImageFormat+CLSID.swift

```diff
@@ -0,0 +1,266 @@
+  ///   passed to it.
+  private static let _pngCLSID = _computeCLSID(forPathExtension: "png")!
+
+  /// The `CLSID` value corresponding to the JPEG image format.
+  ///
+  /// - Note: The named constant [`ImageFormatJPEG`](https://learn.microsoft.com/en-us/windows/win32/gdiplus/-gdiplus-constant-image-file-format-constants)
+  ///   is not the correct value and will cause `Image::Save()` to fail if
+  ///   passed to it.
+  private static let _jpegCLSID = _computeCLSID(forPathExtension: "jpg")!
+
+  /// The `CLSID` value corresponding to this image format.
+  public var clsid: CLSID {
+    switch kind {
+    case .png:
+      Self._pngCLSID
+    case .jpeg:
+      Self._jpegCLSID
+    case let .systemValue(clsid):
+      clsid as! CLSID
+    }
+  }
+
+  /// Construct an instance of this type with the given `CLSID` value and
+  /// encoding quality.
+  ///
+  /// - Parameters:
+  ///   - clsid: The `CLSID` value corresponding to the image format to use when
+  ///     encoding images.
+  ///   - encodingQuality: The encoding quality to use when encoding images. For
+  ///     the lowest supported quality, pass `0.0`. For the highest supported
+  ///     quality, pass `1.0`.
+  ///
+  /// If the target image format does not support variable-quality encoding,
+  /// the value of the `encodingQuality` argument is ignored.
+  ///
+  /// If `clsid` does not represent an image format supported by GDI+, the
+  /// result is undefined. For a list of image formats supported by GDI+, see
+  /// the [GetImageEncoders()](https://learn.microsoft.com/en-us/windows/win32/api/gdiplusimagecodec/nf-gdiplusimagecodec-getimageencoders)
+  /// function.
+  public init(_ clsid: CLSID, encodingQuality: Float = 1.0) {
+    self.init(kind: .systemValue(clsid), encodingQuality: encodingQuality)
+  }
+
+  /// Construct an instance of this type with the given path extension and
+  /// encoding quality.
+  ///
+  /// - Parameters:
+  ///   - pathExtension: A path extension corresponding to the image format to
+  ///     use when encoding images.
+  ///   - encodingQuality: The encoding quality to use when encoding images. For
+  ///     the lowest supported quality, pass `0.0`. For the highest supported
+  ///     quality, pass `1.0`.
+  ///
+  /// If the target image format does not support variable-quality encoding,
+  /// the value of the `encodingQuality` argument is ignored.
+  ///
+  /// If `pathExtension` does not correspond to an image format supported by
+  /// GDI+, this initializer returns `nil`. For a list of image formats
+  /// supported by GDI+, see the [GetImageEncoders()](https://learn.microsoft.com/en-us/windows/win32/api/gdiplusimagecodec/nf-gdiplusimagecodec-getimageencoders)
+  /// function.
+  public init?(pathExtension: String, encodingQuality: Float = 1.0) {
+    let pathExtension = pathExtension.drop { $0 == "." }
+    let clsid = Self._computeCLSID(forPathExtension: String(pathExtension))
+    if let clsid {
```

Comment:
Would a `guard` be better?

Reply:
Doesn't make a huge difference. I can change it if you feel strongly.